### DISCUSSION TITLE
Parsing `covered=*` and using it in the pedestrian cost calculation

### DISCUSSION THREAD

**Discussion by LZhaoM** [valhalla/valhalla#4744] [Ideas]
Hi,

I am wondering if it is worth parsing the key `covered=*` from OSM data and adding an option for using this tag in the pedestrian cost calculation (just like `use_lit`). It may be useful for pedestrians when it is raining outside, or when pedestrians want to avoid sunlight.

Thank you for reading.

references:
[https://wiki.openstreetmap.org/wiki/Key:covered](url)

**Comment by chrstnbwnkl** [comment]
This one goes right on the pile of great ideas that are difficult or impossible to implement right now :smile:  the problem is that `DirectedEdge` is running out of spare bits to set this kind of information on, so once a v4 release comes around and some more space is reserved on that structure, this could be implemented fairly easily.
